service/src/main.rs

```diff
@@ -1,229 +1,14 @@
-
-        let attestable = response
-            .headers()
-            .get("graph-attestable")
-            .map_or(false, |value| {
-                value.to_str().map(|value| value == "true").unwrap_or(false)
-            });
-
-        let body = response
-            .text()
-            .await
-            .map_err(SubgraphServiceError::QueryForwardingError)?;
-
-        Ok((request, SubgraphServiceResponse::new(body, attestable)))
-    }
-}
-
-/// Run the subgraph indexer service
 #[tokio::main]
-async fn main() -> Result<(), Error> {
-    tracing_subscriber::fmt::init();
-
-    // Parse command line and environment arguments
-    let cli = Cli::parse();
-
-    // Load the json-rpc service configuration, which is a combination of the
-    // general configuration options for any indexer service and specific
-    // options added for JSON-RPC
-    let config = match Config::load(&cli.config) {
-        Ok(config) => config,
-        Err(e) => {
-            error!(
-                "Invalid configuration file `{}`: {}",
-                cli.config.display(),
-                e
-            );
-            std::process::exit(1);
-        }
-    };
-
-    // Parse basic configurations
-    build_info::build_info!(fn build_info);
-    let release = IndexerServiceRelease::from(build_info());
-
-    // Some of the subgrpah service configuration goes into the so-called
-    // "state", which will be passed to any request handler, middleware etc.
-    // that is involved in serving requests
-    let state = Arc::new(SubgraphServiceState {
-        config: config.clone(),
-        database: database::connect(&config.common.database.postgres_url).await,
-        cost_schema: routes::cost::build_schema().await,
-        graph_node_client: reqwest::ClientBuilder::new()
-            .tcp_nodelay(true)
-            .timeout(Duration::from_secs(30))
-            .build()
-            .expect("Failed to init HTTP client for Graph Node"),
-        graph_node_status_url: config
-            .common
-            .graph_node
-            .as_ref()
-            .expect("Config must have `common.graph_node.status_url` set")
-            .status_url
-            .clone(),
-        graph_node_query_base_url: config
-            .common
-            .graph_node
-            .as_ref()
-            .expect("config must have `common.graph_node.query_url` set")
-            .query_base_url
-            .clone(),
-    });
-
-    IndexerService::run(IndexerServiceOptions {
-        release,
-        config: config.common.clone(),
-        url_namespace: "subgraphs",
-        metrics_prefix: "subgraph",
-        service_impl: SubgraphService::new(state.clone()),
-        extra_routes: Router::new()
-            .route("/cost", post(routes::cost::cost))
-            .route("/status", post(routes::status))
-            .with_state(state),
-    })
-    .await
+async fn main() {
+    if let Err(e) = run().await {
+        eprintln!("Indexer service error: {e}");
```

Comment:
Why not move the `tracing_subscriber` initialization here, and use `tracing:error` instead?

Reply:
This is also inspired by the documentation on `std::process::exit` that I've quoted in response to your previous comment. If we reach this error path everything by this point has been dropped out of scope except for the error that's bubbled up. Interested to hear what you think!

Reply:
If we move `tracing_subscriber::fmt::init();` at the start of main, and use `ExitCode` + `return` instead of calling for early termination, we should avoid all of those potential pitfalls, I think.